rekor-v2-spec.md

```diff
@@ -0,0 +1,241 @@
+
+#### 4.1.4 Canonicalization
+
+The transparency service will perform type-specific canonicalization on records
+to de-duplicate equivalent entries. An equivalent entry is one that represents
+the same signing event. Sources of duplication include encoding schemes of the
+public key or signature. For example, the transparency service must de-duplicate
+an entry that encodes the public key with DER-encoding from one that uses the
+PEM-encoding. Canonicalization must not remove any necessary information to
+validate the signature. The service must store enough information in the Merkle
+tree leaf for an interested party to perform the same record validation to
+ensure entry auditability.
+
+### 4.2 HTTP and gRPC API
+
+The Rekor v2 write API is accessible over HTTP or gRPC, and the read-only tiles
+API and checkpoint are accessible over HTTP. Refer to the [protobuf
+specification](https://github.com/sigstore/rekor-tiles/tree/main/api/proto) for
+the exact schema, and the [clients
+guide](https://github.com/sigstore/rekor-tiles/blob/main/CLIENTS.md) for
+examples.
+
+### 4.3 Checkpoint Format
+
+The Merkle tree checkpoint, accessible via a GET request to the
+`/api/v2/checkpoint` endpoint, follows the [C2SP Transparency Log
+Checkpoints](https://github.com/C2SP/C2SP/blob/main/tlog-checkpoint.md)
+specification. The checkpoint MAY additionally include witness signatures as
+specified in the [Transparency Log Witness
+Protocol](https://github.com/C2SP/C2SP/blob/main/tlog-witness.md).
+
+## 5. Client Workflows
+
+### 5.1 Signers
+
+A client signing an artifact uploads the entry record, formatted as either a
+[HashedRekordRequestV002](https://github.com/sigstore/rekor-tiles/blob/f3cba09c2f92f1d2d5a7ca2b5694b66941a82d41/api/proto/rekor/v2/hashedrekord.proto#L31C9-L31C32)
+or
+[DSSERequestV002](https://github.com/sigstore/rekor-tiles/blob/f3cba09c2f92f1d2d5a7ca2b5694b66941a82d41/api/proto/rekor/v2/dsse.proto#L32C9-L32C24) to the
+`/api/v2/log/entries` (HTTP) or `dev.sigstore.rekor.v2.Rekor.CreateEntry` (gRPC)
+service. The client receives, as a response, a
+[TransparencyLogEntry](https://github.com/sigstore/protobuf-specs/blob/4df5baadcdb582a70c2bc032e042c0a218eb3841/protos/sigstore_rekor.proto#L94),
+which contains the latest checkpoint and an inclusion proof for the entry.
+Because entries are batched before being integrated, and because checkpoints are
+published only at regular intervals, the client may have to wait for a few
+seconds before receiving a response, depending on the configuration of the
+service.
+
+### 5.2 Artifact Verifiers
+
+A client verifying an artifact MUST NOT perform an online lookup of the entry.
+The signing client must provide the entry's checkpoint, inclusion proof,
+verification key and digest in the verification material it provides for the
+verifying client. The verifying client uses the material it is provided with and
+the public key of the log to verify the entry's inclusion in the log.
+
+### 5.3 Monitors
```

Comment:
Similarly, "monitor" is overloaded. Can we say this is a "verifier for log entries, e.g. log monitor"?

Reply:
I kind of think the benefit of a specification document is that we get to decide what means what. I think monitor is the best term for the function that rekor-monitor does, i.e. monitoring for identity usage.

Reply:
In the terminology section and/or here, can we mention that monitors are referred to as "verifiers" in the claimant model? Just want to try to align to that terminology since we have published a [claimant model](https://github.com/sigstore/community/tree/main/docs/claimantmodel#rekor-identity-based-signature). I know these terms aren't widely known though, so I tend to redefine them whenever I use them. Can we also define what a Believer is in the above section?

Reply:
Oh goodness that's confusing. Yes added to both.